flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -682,6 +686,29 @@ public Table fromCall(Class<? extends UserDefinedFunction> function, Object... a
                 operationTreeBuilder.tableFunction(Expressions.call(function, arguments)));
     }
 
+    @Override
+    public Model fromModelPath(String modelPath) {
+        UnresolvedIdentifier unresolvedIdentifier = getParser().parseIdentifier(modelPath);
+        ObjectIdentifier modelIdentifier = catalogManager.qualifyIdentifier(unresolvedIdentifier);
+        return catalogManager
+                .getModel(modelIdentifier)
+                .map(this::createModel)
+                .orElseThrow(
+                        () ->
+                                new ValidationException(
```

Comment:
throw ModelNotExistException?

Reply:
ModelNotExistException is for catalog API only